cluster-autoscaler/core/scale_up.go

```diff
@@ -467,6 +475,12 @@ func ScaleUp(context *context.AutoscalingContext, processors *ca_processors.Auto
 					continue
 				}
 				nodeInfos[nodeGroup.Id()] = nodeInfo
+
+				option := computeExpansionOption(context, podEquivalenceGroups, nodeGroup, nodeInfo, upcomingNodes)
+
+				if len(option.Pods) > 0 && option.NodeCount > 0 {
```

Comment:
Do we need this if? Why not put it always in the map. Even if pods are empty. No harm, right?

Reply:
Edit - maybe this is safer. We are checking for size of expansion options later. Keep it.